assigner/baserepo.py

```diff
@@ -253,6 +253,28 @@ def list_commits(self, ref_name="master"):
             "/projects/{}/repository/commits".format(self.id), params
         )
 
+    def list_pushes(self):
+        return self._gl_get(
+            "/projects/{}/events?action=pushed".format(self.id)
+        )
+
+    def get_last_HEAD_commit(self, ref="master"):
+        matching_pushes = list(filter(lambda push: push['push_data']['ref'] == ref, self.list_pushes()))
+        commits = self.list_commits(ref)
+
+        HEAD = None
+
+        if commits:
```

Comment:
np: let's flip this to reduce the block's indentation / line length.

```python
if not commits:
    return None
```

Reply:
Can you tell I've been writing rust and dreaming of Option/Result combinators lately?

Reply:
Haha maybe ;)